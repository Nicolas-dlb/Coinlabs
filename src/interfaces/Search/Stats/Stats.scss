.stats {
  display: flex;
  flex-direction: column;
  width: 80%;
  height: 100%;
  margin-right: 2vw;

  .portfolio_graph_container {
    width: 100%;
    height: 87%;
    background: var(--third-bg-color);
    display: flex;
    flex-direction: column;
    align-items: space-between;
    justify-content: flex-start;
    border-radius: 20px;

    .graph {
      top: 17vh;
      left: 1vw;

      &::after {
        top: -20vh;
        width: 94%;
      }
      .grid {
        display: none;
      }
      .recharts-responsive-container {
        position: relative;
        top: -2vh !important;
        right: 1.5vw;
      }
    }
  }
  .tabs {
    display: flex;
    justify-content: space-between;
    align-items: flex-start;
    padding: 16px 22px;
    height: 40px;
    z-index: 10;
    .tabs_left {
      display: flex;
      justify-content: space-between;
      align-items: center;
      width: 50%;
      .btn_portfolio {
        border-radius: 4px;
        padding: 5px 15px;
        background: #2b263f;
        width: 23px;
        text-align: center;
        font-size: 15px;

        &:hover {
          background: var(--primary-font-color);
          color: #2d2a38 !important;
        }
      }
      span {
        cursor: pointer;
        color: #c4c3c9;
        &:hover {
          color: #fff;
        }
      }
      .btn_portfolio_active {
        color: black;
        background: var(--primary-font-color);
      }
    }
    .toggle-select {
      height: 190px !important;
    }
    .tabs_right {
      display: flex;
      align-items: flex-start;
      cursor: pointer;
      color: #f8f7f9;

      border-radius: 6px;
      padding: 3px 0px;
      font-size: 15px;
      z-index: 1000;
      background: #2b263f;
      height: 23px;
      transition: all 0.35s ease-in;
      overflow: hidden;
      width: 103px;
      svg {
        fill: #c8c6d2;
        height: 23px;
        position: relative;
        top: 2px;
        left: 70px;
        margin-left: 5px;
        width: auto;
        pointer-events: none;
      }
    }
  }
}
