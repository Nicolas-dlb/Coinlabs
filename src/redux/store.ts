import { configureStore } from "@reduxjs/toolkit";
import marketReducer from "./reducers/marketSlice";
import walletsReducer from "./reducers/walletsSlice";
import appReducer from "./reducers/appSlice";
import userReducer from "./reducers/userSlice";

export default configureStore({
  reducer: {
    market: marketReducer,
    wallets: walletsReducer,
    app: appReducer,
    user: userReducer,
  },
});
