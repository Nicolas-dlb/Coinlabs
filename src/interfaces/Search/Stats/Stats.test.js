import React from "react";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";
import store from "redux/store";
import { Provider } from "react-redux";
import Stats from "./Stats";

describe("Stats", () => {
  let StatsComponent;
  beforeEach(async () => {
    await act(async () => {
      StatsComponent = await mount(
        <Provider store={store}>
          <Stats name="Bitcoin" time="Week" />
        </Provider>
      );
    });
  });
  it("should render correctly", () => {
    expect(StatsComponent).toMatchSnapshot();
  });
});
