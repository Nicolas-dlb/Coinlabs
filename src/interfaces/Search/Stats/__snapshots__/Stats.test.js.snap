// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats should render correctly 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <pure(Stats)
    name="Bitcoin"
    time="Week"
  >
    <Stats
      name="Bitcoin"
      time="Week"
    >
      <div
        className="portfolio_stats"
      >
        <p
          className="title"
        >
          Portfolio stats
        </p>
        <div
          className="portfolio_graph_container"
        >
          <div
            className="tabs"
          >
            <div
              className="tabs_left"
            >
              <span
                className="btn_portfolio"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                1w
              </span>
              <span
                className="btn_portfolio btn_portfolio_active"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                1m
              </span>
              <span
                className="btn_portfolio"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                3m
              </span>
              <span
                className="btn_portfolio"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                6m
              </span>
              <span
                className="btn_portfolio"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                1y
              </span>
            </div>
            <div
              className="tabs_right"
              id="select-c"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <Select
                items="crypto2"
                setCryptoSelected={[Function]}
              >
                <div>
                  <select
                    className="crypto2 exchange_element"
                  >
                    <option
                      className="option exchange_element"
                      label="All"
                      value="All"
                    >
                      All
                    </option>
                    <option
                      className="option exchange_element"
                      label="Bitcoin"
                      value="Bitcoin"
                    >
                      Bitcoin
                    </option>
                    <option
                      className="option exchange_element"
                      label="Ethereum"
                      value="Ethereum"
                    >
                      Ethereum
                    </option>
                    <option
                      className="option exchange_element"
                      label="Ripple"
                      value="Ripple"
                    >
                      Ripple
                    </option>
                    <option
                      className="option exchange_element"
                      label="Litecoin"
                      value="Litecoin"
                    >
                      Litecoin
                    </option>
                    <option
                      className="option exchange_element"
                      label="Neo"
                      value="Neo"
                    >
                      Neo
                    </option>
                  </select>
                  <div
                    className="crypto2-select exchange_element"
                  >
                    <svg
                      className="select-crypto2-arrow exchange_element"
                      fill="#605b95"
                      height="24px"
                      viewBox="0 0 24 24"
                      width="24px"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0V0z"
                        fill="none"
                      />
                      <path
                        d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                      />
                    </svg>
                    <button
                      aria-label="btn-select"
                      className="btn-crypto2 exchange_element"
                      id="btn-select"
                      onClick={[Function]}
                      type="button"
                      value="All"
                    />
                    <div
                      className="crypto2-list exchange_element"
                      id="crypto2-list"
                    >
                      <ul
                        id="crypto2-items"
                      />
                    </div>
                  </div>
                </div>
              </Select>
            </div>
          </div>
          <Graph
            color="Green"
            data={
              Array [
                Object {
                  "Price": 0,
                  "timestamp": 3227632,
                  "tokens": 0,
                },
              ]
            }
            height={230}
            selected="Bitcoin"
            tooltip={true}
            width={0}
          >
            <div
              className="graph_container"
            >
              <div
                className="graph"
              >
                <ResponsiveContainer
                  debounce={0}
                  height={230}
                  width="90%"
                >
                  <ResizeDetector
                    handleHeight={true}
                    handleWidth={true}
                    onResize={[Function]}
                    targetRef={
                      Object {
                        "current": <div
                          class="recharts-responsive-container"
                          style="width: 90%; height: 230px;"
                        />,
                      }
                    }
                  >
                    <div
                      className="recharts-responsive-container"
                      style={
                        Object {
                          "height": 230,
                          "maxHeight": undefined,
                          "minHeight": undefined,
                          "minWidth": undefined,
                          "width": "90%",
                        }
                      }
                    />
                  </ResizeDetector>
                </ResponsiveContainer>
              </div>
            </div>
          </Graph>
        </div>
      </div>
    </Stats>
  </pure(Stats)>
</Provider>
`;
